db/post_migrate/20220825005115_backfill_outbound_message_id.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class BackfillOutboundMessageId < ActiveRecord::Migration[7.0]
+  def up
+    # best effort backfill, we don't care about years worth of message_id
+    # preservation
+    #
+    # we also don't need to backfill outbound_message_id for posts that
+    # do _not_ have an incoming email linked, since that will be backfilled
+    # at runtime if it is missing
+    sql_query = <<~SQL
+      UPDATE posts
+      SET outbound_message_id = ie.message_id
+      FROM incoming_emails AS ie
```

Comment:
We can try out this query first but if we see lock timeouts, we might have to break this update query down into smaller sub queries.

Reply:
Yep sure, I'm hoping this is not a huge amount of records for most sites.